cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverPluginTest.java

```diff
@@ -262,17 +308,31 @@ public Long call() throws Exception {
     }, 120, TimeUnit.SECONDS);
   }
 
-  private void resetLogSaverPluginCheckpoint(long offset) throws Exception {
+  private void resetLogSaverPluginCheckpoint(List<LogEvent> allEvents, long offset, long mismatchOffset)
+    throws Exception {
+    // If offset is -1, set targetTime as 0. If offset is smaller than the size of allEvents,
```

Comment:
don't use special values to change the logic of a method. It makes it very difficult to understand what is going on. Just pass in the offset and timestamp you want to use for the checkpoint.